src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -182,6 +186,7 @@ CHIP_ERROR OTAImageProcessorImpl::Apply()
                 PlatformMgr().HandleServerShuttingDown();
                 k_msleep(CHIP_DEVICE_CONFIG_SERVER_SHUTDOWN_ACTIONS_SLEEP_MS);
 #ifdef CONFIG_DFU_TARGET_SUIT
+                SetSoftwareRebootReason(SoftwareRebootReason::kSoftwareUpdate);
                 dfu_target_suit_reboot();
```

Comment:
Just wonder.. Maybe we can call `Reboot(SoftwareRebootReason::kSoftwareUpdate);` and inside Reboot.cpp call `dfu_target_suit_reboot();` for `kSoftwareUpdate`?

@ArekBalysNordic what do you think?